File: Selenium-java/src/test/java/LoginTest.java
import org.junit.jupiter.api.*;
import org.openqa.selenium.*;
import org.openqa.selenium.chrome.ChromeDriver;
import java.nio.file.Paths;

public class LoginTest {
    private WebDriver driver;

    @BeforeEach
    public void setUp() {
        // TODO: configure WebDriver location or use WebDriverManager
        driver = new ChromeDriver();
    }

    @AfterEach
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }

    @Test
    public void endToEndLogin() {
        String fileUrl = Paths.get("../public/login.html").toUri().toString();
        driver.get(fileUrl);

        // TODO: Fill in username and password fields
        // driver.findElement(By.id("username")).sendKeys("your-username");
        // driver.findElement(By.id("password")).sendKeys("your-password");
        // driver.findElement(By.id("submit")).click();

        // TODO: Assert welcome message is displayed
        // WebElement message = driver.findElement(By.id("welcome-text"));
        // Assertions.assertEquals("Welcome back, your-username!", message.getText());
    }
}

File: Selenium-javascript/tests/login.test.js
const {Builder, By} = require('selenium-webdriver');
const path = require('path');
const assert = require('assert');

describe('end-to-end login', function() {
  this.timeout(30000);
  let driver;

  before(async () => {
    driver = new Builder().forBrowser('chrome').build();
  });

  after(async () => {
    if (driver) {
      await driver.quit();
    }
  });

  it('logs in successfully', async () => {
    const filePath = path.resolve(__dirname, '../../public/login.html');
    await driver.get('file://' + filePath);

    // TODO: Fill in the username and password fields
    // await driver.findElement(By.id('username')).sendKeys('your-username');
    // await driver.findElement(By.id('password')).sendKeys('your-password');
    // await driver.findElement(By.id('submit')).click();

    // TODO: Verify welcome message
    // const text = await driver.findElement(By.id('welcome-text')).getText();
    // assert.strictEqual(text, 'Welcome back, your-username!');
  });
});
